### PR TITLE
Add Move Top/Bottom to playlist

### LINKED ISSUE #4603: [Feature Request]: Add Move Top/Bottom in the playlist
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

Currently there is only move up/down in the playlist. It is sometimes troublesome when needed to arrange the playlist and needed more clicks to move the video to the top/bottom of the playlist

### Proposed Solution

Add button for the Move to Top/Bottom of the playlist

### Alternatives Considered

- 

### Issue Labels

improvement to existing feature

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ArthurKun21**
# Add Move Top/Bottom to playlist

## Pull Request Type

- [ ] Bugfix
- [X] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue

closes #4603

## Description

This adds another button to move to Top/Bottom of the playlist

## Screenshots 

electron_xwa9V0JBLI

## Testing 

1. Add at least 3 videos to the playlist
2. Go to the 2nd Video, click on the Move Video Top and it should move to the Top of the playlist
3. Go to the 2nd Video, click on the Move Video Bottom and it should move to the Bottom of the playlist

## Desktop

- **OS: Window**
- **OS Version: Windows 11**
- **FreeTube version: 0.19.1**

## Additional context

This is my first time messing around with javascript/vue. Please forgive some errors if there is any.

### CODE REVIEW

**Comment by jasonhenriquez** [comment]
Thanks for the initiative on this @ArthurKun21! My suggestions (haven't looked into the code yet):

1. Change labels to "Move Video to Top" and "Move Video to Bottom".
2. Move "Move Video to Top" arrow to the right of the "Move Video Up/Down" buttons, such that it's next to its counterpart.
3. I like the icon meaning, but I worry the two arrows' different meanings could be confusing to some. If you don't mind, could you try replacing these new icons with `arrow-up-long` and `arrow-down-long`, and possibly even `up-long` and `down-long`, and take screenshots of the result? I'm interested to see if it might be clearer from a distance.

Thanks @ArthurKun21!

**Comment by jasonhenriquez** [comment]
Just had an idea: changing the original up/down to `angle-up` and `angle-down`, and making these new icons `angles-up` and `angles-down`.

File: src/renderer/main.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ import { register as registerSwiper } from 'swiper/element'
 import {
   faAngleDown,
   faAngleUp,
+  faAnglesDown,
+  faAnglesUp,
```
Comment (jasonhenriquez) [inline src/renderer/main.js:17]:
```suggestion
  faAngleDown,
  faAnglesDown,
  faAnglesUp,
  faAngleUp,
```
Alphabetized (see comment at top of file)

File: src/renderer/main.js [hunk 2]
```diff
@@ -93,6 +95,8 @@ library.add(
   // solid icons
   faAngleDown,
   faAngleUp,
+  faAnglesDown,
+  faAnglesUp,
```
Comment (jasonhenriquez) [inline src/renderer/main.js:99]:
```suggestion
  faAngleDown,
  faAnglesDown,
  faAnglesUp,
  faAngleUp,
```
Alphabetized (see comment at top of file)

File: src/renderer/views/Playlist/Playlist.js [hunk 3]
```diff
@@ -330,7 +330,60 @@ export default defineComponent({
...
+        protected: this.selectedUserPlaylist.protected,
+        description: this.playlistDescription,
+        videos: playlistItems,
+        _id: this.playlistId
+      }
+      try {
+        this.updatePlaylist(playlist)
+        this.playlistItems = playlistItems
+      } catch (e) {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["There was an issue with updating this playlist."]'))
```
Comment (jasonhenriquez) [inline src/renderer/views/Playlist/Playlist.js:383]: Needs localization string

File: src/renderer/views/Playlist/Playlist.js [hunk 4]
```diff
@@ -330,7 +330,60 @@ export default defineComponent({
...
+        console.error(e)
+      }
+    },
+    moveVideoBottom: function(videoId, playlistItemId) {
+      const playlistItems = [].concat(this.playlistItems)
+      const videoIndex = playlistItems.findIndex((video) => {
+        return video.videoId === videoId && video.playlistItemId === playlistItemId
+      })
+      if (videoIndex === playlistItems.length - 1) {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["This video cannot be moved to the bottom."]'))
```
Comment (jasonhenriquez) [inline src/renderer/views/Playlist/Playlist.js:366]: Needs localization string

File: src/renderer/views/Playlist/Playlist.js [hunk 5]
```diff
@@ -330,7 +330,60 @@ export default defineComponent({
...
       })
     },
-
+    moveVideoTop: function (videoId, playlistItemId) {
+      const playlistItems = [].concat(this.playlistItems)
+      const videoIndex = playlistItems.findIndex((video) => {
+        return video.videoId === videoId && video.playlistItemId === playlistItemId
+      })
+      if (videoIndex === 0) {
+        showToast(this.$t('User Playlists.SinglePlaylistView.Toast["This video cannot be moved to the top."]'))
```
Comment (jasonhenriquez) [inline src/renderer/views/Playlist/Playlist.js:339]: Needs localization string

File: static/locales/en-US.yaml [hunk 6]
```diff
@@ -152,6 +152,8 @@ User Playlists:
   Add to Favorites: Add to {playlistName}
   Remove from Favorites: Remove from {playlistName}
 
+  Move Video Top: Move Video to Top
+  Move Video Bottom: Move Video to Bottom
```
Comment (jasonhenriquez) [inline static/locales/en-US.yaml:156]: Update the labels to match the new English labels

File: src/renderer/components/ft-list-video/ft-list-video.vue [hunk 7]
```diff
@@ -69,10 +69,19 @@
           :size="appearance === `watchPlaylistItem` ? 14 : 18"
           @click="toggleQuickBookmarked"
         />
+        <ft-icon-button
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-list-video/ft-list-video.vue:72]: Move this button to be adjacent to the `Move Video to Bottom` button

**Comment by jasonhenriquez** [comment]
Left a handful of comments after looking into the code. Thanks @ArthurKun21!

**Comment by PikachuEXE** [comment]
I will have to review this after https://github.com/FreeTubeApp/FreeTube/pull/4597 is merged which introduced pagination for user playlists and test on playlists with > 100 items

**Comment by absidue** [comment]
The video thumbnail looks really crowded now. As move to top and move to bottom are presumably used a lot less often than the normal up and down, maybe these two new options should be added to the 3 dots menu?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I already deleted that playlist but i created a new one without any duplicates an its still doing it.


https://github.com/FreeTubeApp/FreeTube/assets/73130443/5b026bc4-241c-472b-a290-19e9aa278ebd

**Pull request closed without merging**